### ISSUE TITLE
no-typos rule breaks with assigning functions on this

### ISSUE THREAD

**Issue by maxlapides** [yannickcr/eslint-plugin-react#1378]
With `react/no-typos` enabled, linting this code

```javascript
function test() {
  this.foo = foo
  function foo() { console.log('oh noes') }
}
```

Will throw this error:

```
Cannot read property 'type' of undefined
TypeError: Cannot read property 'type' of undefined
    at looksLikeExport ([my-project]/node_modules/eslint/lib/util/source-code.js:78:19)
    at SourceCode.getJSDocComment ([my-project]/node_modules/eslint/lib/util/source-code.js:311:21)
    at Object.isExplicitComponent ([my-project]/node_modules/eslint-plugin-react/lib/util/Components.js:218:34)
    at Object.isES6Component ([my-project]/node_modules/eslint-plugin-react/lib/util/Components.js:201:17)
    at Linter.MemberExpression ([my-project]/node_modules/eslint-plugin-react/lib/rules/no-typos.js:73:18)
    at emitOne (events.js:101:20)
    at Linter.emit (events.js:188:7)
    at NodeEventGenerator.applySelector ([my-project]/node_modules/eslint/lib/util/node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors ([my-project]/node_modules/eslint/lib/util/node-event-generator.js:294:22)
    at NodeEventGenerator.enterNode ([my-project]/node_modules/eslint/lib/util/node-event-generator.js:308:14)
    at CodePathAnalyzer.enterNode ([my-project]/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:602:23)
    at Traverser.enter ([my-project]/node_modules/eslint/lib/linter.js:925:36)
```

**Comment by ljharb** [comment]
This might be fixed by #1372 and #1375.

**Comment by jseminck** [comment]
Definitely fixed. Just tested it through unit tests.
